### PR TITLE
rewrite stylesheet management

### PR DESCRIPTION
**Pull request by ax-sc**

### CODE REVIEW

**Comment by 3j14** [comment]
The failing test fails quite randomly and is nothing to be concerned about. I suspect that redis is not fast enough to keep up with the newly created session. I will have to check this later.

File: doku/models/schemas/template.py [hunk 1]
```diff
@@ -63,20 +63,36 @@ def update(cls, commit=True):
...
 
     id = auto_field()
     name = NotEmptyString()
     source = auto_field()
-    base_templates = Nested("TemplateSchema", exclude=("base_style",), many=True)
+    base_templates = Nested("TemplateSchema", exclude=("base_style","styles"), many=True)
     templates = Nested("TemplateSchema", exclude=("styles",), many=True)
 
     upload_url = fields.Method("_upload_url", dump_only=True, allow_none=True)
+    delete_url = fields.Method("_delete_url", dump_only=True, allow_none=True)
```
Comment (3j14) [inline doku/models/schemas/template.py]: The `delete_url` is already defined and does not need to be overwritten. 

https://github.com/desklab/doku/blob/92bbfecc7e5c06cbf9b6b5ec67b4af15f9eabdc3/doku/models/schemas/common.py#L55

It's enough to just overwrite `def delete(. . .)`.

File: doku/static/src/store/types/actions.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ export const ADD_STYLESHEET_TO_TEMPLATE = 'addStylesheet';
 export const REMOVE_STYLESHEET_FROM_TEMPLATE = 'removeStylesheet';
 export const UPLOAD_STYLESHEET = 'uploadStylesheet';
 export const UPDATE_STYLESHEET = 'updateStylesheet';
+export const DELETE_STYLESHEET = 'deleteStylesheet';
```
Comment (3j14) [inline doku/static/src/store/types/actions.js:16]: We use **remove** instead of **delete** in most places ;)

**Review by 3j14 — APPROVED** [review]
Just a few cosmetic things, other than that this already looks promising

**Comment by ax-sc** [comment]
> Just a few cosmetic things, other than that this already looks promising

There are still some things not working properly, for example uploading/adding new stylesheets and selecting them in a template. I´m going to notify you once i´m finished.

File: doku/static/src/store/types/actions.js (hunk 2 above)
Comment (ax-sc) [reply doku/static/src/store/types/actions.js:16]: Maybe we should overthink this as we are also *removing* stylesheets from templates without deleting them?

File: doku/models/schemas/template.py (hunk 1 above)
Comment (ax-sc) [reply doku/models/schemas/template.py]: fixed in 18ca3d14a8ab0fc53f87d93a1b0fb2397506a635

**Comment by ax-sc** [comment]
@jnsdrtlf I think this is now ready to be merged. May we should remove the `StylesheetModal` component as it is not used anymore?

**Comment by 3j14** [comment]
Can we merge this?

**Comment by 3j14** [comment]
Sorry, haven't seen your comment. Sure thing, you can just delete it. Could you also make sure to move template and stylesheet related files into their own separate folder alongside the `ui` folder, just like I did with `variables`? This helps reduce the number of files inside the `ui` folder, which becomes very confusion.

**Review by 3j14 — APPROVED** [review]

**Pull request merged**